diff --git a/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php b/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php
index d4945cf3..e96b71d6 100644
--- a/SqlContentEntityStorage.php
+++ b/SqlContentEntityStorage.php
@@ -808,11 +808,14 @@ class SqlContentEntityStorage extends ContentEntityStorageBase implements SqlEnt
       if ($update) {
         $default_revision = $entity->isDefaultRevision();
         if ($default_revision) {
-          $this->database
-            ->update($this->baseTable)
-            ->fields((array) $record)
-            ->condition($this->idKey, $record->{$this->idKey})
-            ->execute();
+		  $fields = (array)$record;
+		  unset($fields[$this->idKey]);
+
+		  $this->database
+			->update($this->baseTable)
+			->fields($fields)
+			->condition($this->idKey, $record->{$this->idKey})
+			->execute();
         }
         if ($this->revisionTable) {
           if ($full_save) {
@@ -821,9 +824,13 @@ class SqlContentEntityStorage extends ContentEntityStorageBase implements SqlEnt
           else {
             $record = $this->mapToStorageRecord($entity->getUntranslated(), $this->revisionTable);
             $entity->preSaveRevision($this, $record);
+			
+            $fields = (array)$record;
+            unset($fields[$this->revisionKey]);
+
             $this->database
               ->update($this->revisionTable)
-              ->fields((array) $record)
+              ->fields($record)
               ->condition($this->revisionKey, $record->{$this->revisionKey})
               ->execute();
           }
@@ -1053,9 +1060,12 @@ class SqlContentEntityStorage extends ContentEntityStorageBase implements SqlEnt
       }
     }
     else {
+      $fields = (array)$record;
+      unset($fields[$this->revisionKey]);
+
       $this->database
         ->update($this->revisionTable)
-        ->fields((array) $record)
+        ->fields($fields)
         ->condition($this->revisionKey, $record->{$this->revisionKey})
         ->execute();
     }
