diff --git a/src/RequestHandler.php b/src/RequestHandler.php
index 73bc2c4..5f0b903 100644
--- a/src/RequestHandler.php
+++ b/src/RequestHandler.php
@@ -5,6 +5,8 @@ namespace Drupal\mm_rest;
 use Drupal\Core\Render\RenderContext;
 use Drupal\Core\Routing\RouteMatchInterface;
 use Drupal\rest\ResourceResponse;
+use Drupal\rest\RequestHandler as CoreRequestHandler;
+use Drupal\rest\RestResourceConfigInterface;
 use Symfony\Component\DependencyInjection\ContainerAwareInterface;
 use Symfony\Component\DependencyInjection\ContainerAwareTrait;
 use Symfony\Component\HttpFoundation\Request;
@@ -16,83 +18,20 @@ use Symfony\Component\Serializer\Exception\UnexpectedValueException;
 /**
  * Acts as intermediate request forwarder for resource plugins.
  */
-class RequestHandler implements ContainerAwareInterface {
-
-  use ContainerAwareTrait;
+class RequestHandler extends CoreRequestHandler {

   /**
-   * Handles a web API request.
-   *
-   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
-   *   The route match.
-   * @param \Symfony\Component\HttpFoundation\Request $request
-   *   The HTTP request object.
-   *
-   * @return \Symfony\Component\HttpFoundation\Response|array|object
-   *   The response object.
+   * @inheritdoc
    */
-  public function handle(RouteMatchInterface $route_match, Request $request) {
-
-    $plugin = $route_match->getRouteObject()->getDefault('_plugin');
-    $method = strtolower($request->getMethod());
-
-    $resource = $this->container
-      ->get('plugin.manager.rest')
-      ->getInstance(array('id' => $plugin));
-
-    // Deserialize incoming data if available.
-    $serializer = $this->container->get('serializer');
-    $received = $request->getContent();
-    $unserialized = NULL;
-    // @todo Move de-serialization to an Event Subscriber
-    if (!empty($received)) {
-      $format = $request->getContentType();
-
-      // Only allow serialization formats that are explicitly configured. If no
-      // formats are configured allow all and hope that the serializer knows the
-      // format. If the serializer cannot handle it an exception will be thrown
-      // that bubbles up to the client.
-      $config = $this->container->get('config.factory')->get('rest.settings')->get('resources');
-      $method_settings = $config[$plugin][$request->getMethod()];
-      if (empty($method_settings['supported_formats']) || in_array($format, $method_settings['supported_formats'])) {
-        $definition = $resource->getPluginDefinition();
-        $class = $definition['serialization_class'];
-        try {
-          $unserialized = $serializer->deserialize($received, $class, $format, array('request_method' => $method));
-        }
-        catch (UnexpectedValueException $e) {
-          // @todo Throw exception
-          $error['error']['code'] = $e->getCode();
-          $error['error']['message'] = $e->getMessage();
-          $content = $serializer->serialize($error, $format);
-          return new Response($content, 400, array('Content-Type' => $request->getMimeType($format)));
-        }
-      }
-      else {
-        throw new UnsupportedMediaTypeHttpException();
-      }
-    }
-
-    // Determine the request parameters that should be passed to the resource
-    // plugin.
-    $route_parameters = $route_match->getParameters();
-    $parameters = array();
-    // Filter out all internal parameters starting with "_".
-    foreach ($route_parameters as $key => $parameter) {
-      if ($key{0} !== '_') {
-        $parameters[] = $parameter;
-      }
-    }
-
-   // Invoke the operation on the resource plugin.
-    $result = call_user_func_array(array($resource, $method), array_merge($parameters, array($unserialized, $request)));
+  public function handle(RouteMatchInterface $route_match, Request $request, RestResourceConfigInterface $_rest_resource_config) {
+    $response = parent::handle($route_match, $request, $_rest_resource_config);

     // Add rest settings config's cache tags.
     // @todo Add mm_rest config cache tag.
     // @todo Use DI
-    \Drupal::service('mm_rest.cacheable_metadata_collector')->addCacheableDependency($this->container->get('config.factory')->get('rest.settings'));
+    \Drupal::service('mm_rest.cacheable_metadata_collector')->addCacheableDependency($this->configFactory->get('rest.settings'));

-    return $result;
+    return $response;
   }

 }